### DISCUSSION TITLE
Unregistering mobile devices from 2FA

### DISCUSSION THREAD

**Discussion by Defeatomizer** [community/community#15076] [Mobile] [answered]
I have recently installed GitHub app on a mobile device and it got registered in the GitHub Mobile section of 2FA settings. After some time I logged out of the app and uninstalled it, but now each time I log in using a browser I instantly get a suggestion to check a sign-in code/request sent to GitHub app on that device (even despite it no longer has this app installed). Installing the app once again on the same device registered the device second time (only registration date was different), disabling and re-enabling of 2FA copied that second registration.
What I want to see is a button near each mobile device registration that allows you to unregister that device from 2FA (if on one of them GitHub is installed and authorized, don't show this button) or even more management options.
On the side note, as I mentioned that reinstalling of the 2FA or the mobile app copies the device registration, I'd like to know if this is a bug or there's just no way to check if a device is the same as one of registered before.

**Comment by stevepopovich** [comment]
Hi there! Thanks for the feedback. I think I understand what you are asking for. Let me see if I can I help you.

> On the side note, as I mentioned that reinstalling of the 2FA or the mobile app copies the device registration, I'd like to know if this is a bug or there's just no way to check if a device is the same as one of registered before.

We can verify if a copy truly happens. I think if we are truly registering one device as two devices, that would be a bug. Can you navigate to https://github.com/settings/two_factor_authentication/configure and see what devices are registered? If you see the same device twice, we can chat about that.

Furthermore, if you want to see what devices are OAuth capable, you can navigate to https://github.com/settings/developers and go to OAuth Apps and manually revoke stuff if you want.

Does that make sense? I understand your concerns I think, but not sure 100% what we can change in our implemenation to help you.

**Comment by Defeatomizer** [comment]
>Can you navigate to https://github.com/settings/two_factor_authentication/configure and see what devices are registered?

By the moment of this report creation there were 3 devices with the same name (and this is surely the same device), second and third had the same registration date, first was registered 10 days before other two.
A weird thing though - that oldest device got removed from the list, I guess it expired because it hadn't GitHub running on it for a month.
>Furthermore, if you want to see what devices are OAuth capable, you can navigate to https://github.com/settings/developers and go to OAuth Apps and manually revoke stuff if you want.

I don't have any apps in that tab.

**Comment by stevepopovich** [comment]
What platform is this happening on?

**Comment by Defeatomizer** [comment]
A browser on Windows OS.

**Comment by stevepopovich** [comment]
Are the mobile devices involved Android or iOS?

**Comment by Defeatomizer** [comment]
Android.

**Reply by stevepopovich** [reply]
So to confirm, you logged out and still see your device as registered? Also, what does this behavior prevent from doing? Does it break anything?

**Reply by Defeatomizer** [reply]
> So to confirm, you logged out and still see your device as registered?

Precisely.
>Also, what does this behavior prevent from doing? Does it break anything?

Speaking of the app - no. But
>each time I log in using a browser I instantly get a suggestion to check a sign-in code/request sent to GitHub app...

**Reply by stevepopovich** [reply]
Gotcha. Thanks for the super detailed feedback. I will pass this on to the team and we will take a look!

**Comment by mijiang168** [comment]
> I have recently installed GitHub app on a mobile device and it got registered in the GitHub Mobile section of 2FA settings. After some time I logged out of the app and uninstalled it, but now each time I log in using a browser I instantly get a suggestion to check a sign-in code/request sent to GitHub app on that device (even despite it no longer has this app installed). Installing the app once again on the same device registered the device second time (only registration date was different), disabling and re-enabling of 2FA copied that second registration.
> 
> What I want to see is a button near each mobile device registration that allows you to unregister that device from 2FA (if on one of them GitHub is installed and authorized, don't show this button) or even more management options.
> 
> On the side note, as I mentioned that reinstalling of the 2FA or the mobile app copies the device registration, I'd like to know if this is a bug or there's just no way to check if a device is the same as one of registered before.

**Comment by mijiang168** [comment]
> Android.

**Comment by Defeatomizer** [comment, marked as answer]
Just checked: installed GitHub Mobile, authorized it, [approved 2FA in browser from it a couple of times,] signed out from the app, revoked the app access on my account - the device got unregistered.

**Reply by stevepopovich** [reply]
Hm so it is working as intended?

**Reply by Defeatomizer** [reply]
I think so, yeah. Thank you very much!
